### PR TITLE
MRG, ENH: Allow setting tqdm backend

### LINKED ISSUE #8096: Progressbar not visible in Spyder
Performing a cluster based test (unfortunatelly one that takes a lot of time) I no longer see the progressbar in Spyder:
```
stat_fun(H1): min=-5.699470 max=3.482243
Running initial clustering
Found 17 clusters
Permuting 9999 times...
HBox(children=(FloatProgress(value=0.0, max=9999.0), HTML(value='')))
```
this is most likely a result of using auto mode of `tqdm` progressbar, which seems to be unable to tell whether it is in a notebook or ipython console. It would be nice to have some option to control this behavior (`progressbar='text'` maybe?) as now everyone who works in ipython/Spyder will likely have no progressbar, which for long running tasks is a bit inconvenient.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #8096

@mmagnuski want to give it a try?

### CODE REVIEW

**Comment by mmagnuski** [comment]
Great, thanks, I'll take a look tomorrow! Sorry I didn't follow follow through with the issue.

File: mne/utils/tests/test_progressbar.py [hunk 1]
```diff
@@ -25,6 +26,18 @@ def iter_func(a):
             pass
     pytest.raises(Exception, iter_func, ProgressBar(20))
 
+    # Make sure different progress bars can be used
+    with catch_logging() as log, modified_env(MNE_TQDM='tqdm'), \
+            use_log_level('debug'), ProgressBar(np.arange(3)) as pbar:
+        for p in pbar:
+            pass
```
Comment (hoechenberger) [inline mne/utils/tests/test_progressbar.py:33]: I'm not familiar with this part of MNE or TQDM at all; what is this look intended to achieve? :)

Comment (larsoner) [reply mne/utils/tests/test_progressbar.py:33]: When you iterate over the progressbar it emits its messages, so this is just to get it to do that / actually use the object

Comment (hoechenberger) [reply mne/utils/tests/test_progressbar.py:33]: Thanks for the explanation!

**Review by hoechenberger — APPROVED** [review]
Code looks good, I haven't taken the time to actually **test** it, though.

@mmagnuski Merge if happy.

**Comment by mmagnuski** [comment]
@hoechenberger Good, I'll test it today!

**Comment by hoechenberger** [comment]
Friendly poke @mmagnuski 🙃

**Comment by hoechenberger** [comment]
I'll just go ahead and merge this :) @mmagnuski If you have any suggestions, we can address them in a followup PR.

**Comment by hoechenberger** [comment]
Thanks, @larsoner!

**Comment by mmagnuski** [comment]
Oh, sorry, @hoechenberger, forgot about this, and had poor internet connection till now.

**Comment by hoechenberger** [comment]
@mmagnuski No worries, I can relate 😇

**Pull request merged**